src/Application/Features/Participants/Queries/GetParticipantEnrolmentHistory.cs

```diff
@@ -0,0 +1,105 @@
+        {      
+            var db = unitOfWork.DbContext;
+
+            var results = await (from peh in db.ParticipantEnrolmentHistories
+                                 join u in db.Users on peh.CreatedBy equals u.Id
+                                 join plh in db.ParticipantLocationHistories on peh.ParticipantId equals plh.ParticipantId
+                                 join l in db.Locations on plh.LocationId equals l.Id
+                                 where peh.ParticipantId == request.ParticipantId
+                                 select new ParticipantEnrolmentHistoryDto
+                                 {
+                                     Id = peh.Id,
+                                     ActionDate = peh.Created.HasValue
+                                             ? DateOnly.FromDateTime(peh.Created.Value)
+                                                     : null,
+                                     Status = peh.EnrolmentStatus == 4 ? "Archived" : "Active",
+                                     Event = peh.EnrolmentStatus.Name,
+                                     Reason=peh.Reason,
+                                     AdditionalInformation = peh.AdditionalInformation!,
+                                     LocationName = l.Name,
+                                     SupportWorker = u!.DisplayName!,
+                                     SupportWorkerTenantId = u!.TenantId
+                                 }).ToListAsync();
+
+            if (results.Count == 0)
+                return results;
+            
+            bool hideUser = !request.CurrentUser.AssignedRoles.Any(role => AllowedRoles.Contains(role));
+
+            for (int i = 1; i < results.Count; i++)
+            {
+                // Mask support worker if needed
+                if (hideUser && HiddenTenantIds.Contains(results[i].SupportWorkerTenantId))
+                {
+                    results[i].SupportWorker = "CFO User";
+                }
+
+                // Calculate DaysSincePrevious
+                if (i == results.Count - 1)
+                {
+                    // Last row - compare to today
+                    results[i].DaysSincePrevious = results[i].ActionDate.HasValue
+                        ? (DateTime.Today - results[i].ActionDate!.Value.ToDateTime(TimeOnly.MinValue)).Days
+                        : 0;
+                }
+                else if (results[i].ActionDate.HasValue && results[i + 1].ActionDate.HasValue)
+                {
+                    results[i].DaysSincePrevious =
+                        (results[i + 1].ActionDate!.Value.ToDateTime(TimeOnly.MinValue) -
+                         results[i].ActionDate!.Value.ToDateTime(TimeOnly.MinValue)).Days;
+                }
+                else
+                {
+                    results[i].DaysSincePrevious = 0;
+                }
```

Comment:
```suggestion
                var current = results[i];
                var next = i == results.Count - 1
                    ? DateOnly.FromDateTime(DateTime.Today)
                    : results[i + 1].ActionDate;

                current.DaysSincePrevious = (current.ActionDate.HasValue && next.HasValue)
                    ? next.Value.DayNumber - current.ActionDate.Value.DayNumber
                    : 0;
```

A small improvement can be made here, mostly for readability.